pkg/controller/customdomain/customdomain_utils.go

```diff
@@ -128,15 +132,18 @@ func (r *ReconcileCustomDomain) finalizeCustomDomain(reqLogger logr.Logger, inst
 	}, customIngress)
 	if err != nil {
 		reqLogger.Error(err, fmt.Sprintf("Failed to get %s ingresscontroller", instance.Name))
-		return err
-	}
-	err = r.client.Delete(context.TODO(), customIngress)
-	if err != nil {
-		reqLogger.Error(err, fmt.Sprintf("Failed to delete %s ingresscontroller", instance.Name))
-		return err
+	} else {
```

Comment:
Same here return an error if we fail to get the ingresscontroller to try again? Remove the else?

Reply:
Updated to return err if it's not the NotFound error.